### PR TITLE
Magnus

### PR DESCRIPTION
**Pull request by MHoude2**
Implemented the first and third order Magnus correction for gaussian pump and either gaussian or sinc PMF. Third order is separated into Real and Imaginary parts. 

New functions are in a new module. 

New Example is included which goes through the steps of using the new functions. Some comments are also included to make it clear that cubature is finicky.  

Modified the Read.me to include some info about the new module, and also specified that trying this with other functions might not work as it all relies on Cubature.

Modified requirements and setup.py to include cubature for installation.

Modified version to 0.1.4.

Modified changelog to include new changes.

### CODE REVIEW

**Comment by MHoude2** [comment]
Added the CW functionalities and examples. Also modified readme and changlog again to reflect these additions.

**Comment by MHoude2** [comment]
Forgot to upload the list of refractive indices last time. Modified Magnus functions and Example to now properly function

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+    * New function "Magnus3_Im" which calculates the imaginary part of the third order Magnus term.
+    * New function "Magnus1CW_data" which calculates the first order Magnus term for a Continuous Wave pump and disperion relations obtained from data.
+    * New function "Magnus3CW_data_Re" which calculates the real part of the third order Magnus term for a Continuous Wave pump and disperion relations obtained from data.
+    * New function "Magnus3CW_data_Im" which calculates the imaginary part of the third order Magnus term for a Continuous Wave pump and disperion relations obtained from data.
+    * New function "Magnus1CW_fit" which calculates the first order Magnus term for a Continuous Wave pump and disperion relations obtained from fit parameters.
+    * New function "Magnus3CW_fit_Re" which calculates the real part of the third order Magnus term for a Continuous Wave pump and disperion relations obtained from fit parameters.
+    * New function "Magnus3CW_fit_Im" which calculates the imaginary part of the third order Magnus term for a Continuous Wave pump and disperion relations obtained from fit parameters.
```
Comment (nquesada) [inline .github/CHANGELOG.md]:
```suggestion
### New Features
* New Module `magnus.py` 
    * New function `Magnus1` which calculates the first order Magnus term.
    * New function `Magnus3_Re` which calculates the real part of the third order Magnus term.
    * New function `Magnus3_Im` which calculates the imaginary part of the third order Magnus term.
    * New function `Magnus1CW_data` which calculates the first order Magnus term for a Continuous Wave pump and dispersion relations obtained from data.
    * New function `Magnus3CW_data_Re` which calculates the real part of the third order Magnus term for a Continuous Wave pump and dispersion relations obtained from data.
    * New function `Magnus3CW_data_Im` which calculates the imaginary part of the third order Magnus term for a Continuous Wave pump and dispersion relations obtained from data.
    * New function `Magnus1CW_fit` which calculates the first order Magnus term for a Continuous Wave pump and dispersion relations obtained from fit parameters.
    * New function `Magnus3CW_fit_Re` which calculates the real part of the third order Magnus term for a Continuous Wave pump and dispersion relations obtained from fit parameters.
    * New function `Magnus3CW_fit_Im` which calculates the imaginary part of the third order Magnus term for a Continuous Wave pump and dispersion relations obtained from fit parameters.
```

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+
+### New Documentation
+* New notebook "example_magnus.ipynb" which shows how to use the new module for a Gaussian pump and either Gaussian or Sinc PMF.
+* New notebook "example_magnus_cw_data.ipynb" which shows how to use the new set of 'Magnus#CW_" functions using experimental data.
+* New notebook "example_magnus_cw_fit.ipynb" which shows how to the 'Magnus#CW_fit_" given a set of fit parameters of a dispersion relation.
```
Comment (nquesada) [inline .github/CHANGELOG.md]:
```suggestion
* New notebook "example_magnus.ipynb" which shows how to use the new module for a Gaussian pump and either Gaussian or Sinc PMF.
* New notebook `example_magnus_cw_data.ipynb` which shows how to use the new set of  `Magnus#CW_` functions using experimental data.
* New notebook `example_magnus_cw_fit.ipynb` which shows how to the `Magnus#CW_fit_` given a set of fit parameters of a dispersion relation. 
```

File: NeedALight/magnus.py [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+        J1(w): first order magnus term.
+    
+    """
+    J1 = np.sinc(L*(k_o(ws)+k_e(wi)+k_pol-p_k)/(2*np.pi))
+
+
+
+    return J1
+
+def Magnus3CW_data_Re(omega,ws,wi,k_o,k_e,k_pol,p_k,L):
```
Comment (nquesada) [inline NeedALight/magnus.py]: You probably want to pass black in this file.

Also, you want to make the scaling argument as an optional parameter in the functions. That way you don't need to change your tests but you still give freedom to the user without them having to go an open the file.

**Review by nquesada — COMMENTED** [review]
Minor suggestions below.

File: setup.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
     "jupyter",
     "custom-poling",
     "cubature",
+    "tqdm",
```
Comment (nquesada) [inline setup.py:18]: Nice

**Review by nquesada — APPROVED** [review]
Nice work!

**Pull request merged**